vcsclient/fs.go

```diff
@@ -200,6 +197,28 @@ func GetFileWithOptions(fs vfs.FileSystem, path string, opt GetFileOptions) (*Fi
 	return &fwr, nil
 }
 
+// readDir uses the passed vfs.FileSystem to read from starting at the base path. If
+// shouldRecurse is set to true, it will recurse into all sub-folders and return the
+// full sub-tree.
+func readDir(fs vfs.FileSystem, base string, shouldRecurse bool) ([]*TreeEntry, error) {
```

Comment:
Rwvfs.Walkable

Sent from my iPhone

> On May 15, 2015, at 00:14, Gabriel Aszalos <notifications@github.com> wrote:
> 
> In vcsclient/fs.go:
> 
> > @@ -200,6 +197,28 @@ func GetFileWithOptions(fs vfs.FileSystem, path string, opt GetFileOptions) (*Fi
> >  	return &fwr, nil
> >  }
> >  
> > +// readDir uses the passed vfs.FileSystem to read from starting at the base path. If
> > +// shouldRecurse is set to true, it will recurse into all sub-folders and return the
> > +// full sub-tree.
> > +func readDir(fs vfs.FileSystem, base string, shouldRecurse bool) ([]*TreeEntry, error) {
> Totally. Unfortunately, vfs.FileSystem can not be converted fs.FileSystem because it is lacking the Join method - bummer!
> 
> —
> Reply to this email directly or view it on GitHub.
>